lib/proxy/request.rb

```diff
@@ -63,9 +63,14 @@ def http
     private
 
     def http_init
-      http             = Net::HTTP.new(uri.host, uri.port)
-      http.use_ssl     = uri.scheme == 'https'
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = uri.scheme == 'https'
       http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+      http.open_timeout = Proxy::SETTINGS.foreman_http_timeout
```

Comment:
I think the default value for this is nil. I doubt that works.

Reply:
Right, rebased and added 60 as the default.